### PR TITLE
fix(dropdown): refactored remove/adding to array and removed sort function.

### PR DESCRIPTION
**Pull request by mJarsater**
**Describe pull-request**  
Removed the push/splice methods and moved over to functional programming, map/filter and creating new arrays. Also removed the sort function since it did nothing and emptied the array.

**Solving issue**  
Fixes: -

**How to test**  
1. Go to https://deploy-preview-620--tegel-storybook.netlify.app/ (the latest working deploy)
2. Check in Components -> Dropdown
3. Check that the multiselect and default options all works as intended. 

**Screenshots**  

**Additional context**

### CODE REVIEW

**Comment by EmelieLitwin** [comment]
Looks good to me! Good job! 
Doesn't seem like the label text is changeable in storybook though?

File: components/src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -146,7 +148,8 @@ export class Dropdown {
       if (typeof this.textInput !== 'undefined' || this.textInput === null) {
         this.textInput.focus();
       }
-      // TODO: Maybe can be this.open = !this.open ?
+      // TODO: Maybe can be this.open = !this.open ? - Does not work, dont know why?
+      // eslint-disable-next-line no-unused-expressions, @typescript-eslint/no-unused-expressions
       this.open ? !this.open : this.open;
```
Comment (alexandereneroth) [inline components/src/components/dropdown/dropdown.tsx:153]: I don't think this does anything? I don't see an assignment here.

File: tegel/src/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -146,7 +149,8 @@ export class Dropdown {
       if (typeof this.textInput !== 'undefined' || this.textInput === null) {
         this.textInput.focus();
       }
-      // TODO: Maybe can be this.open = !this.open ?
+      // TODO: Maybe can be this.open = !this.open ? - Does not work, dont know why?
+      // eslint-disable-next-line no-unused-expressions, @typescript-eslint/no-unused-expressions
       this.open ? !this.open : this.open;
```
Comment (alexandereneroth) [inline tegel/src/components/dropdown/dropdown.tsx:154]:
```suggestion
```
Same here, this doesn't appear to do anything

**Review by alexandereneroth — APPROVED** [review]
Nice work

File: components/src/components/dropdown/dropdown.tsx (hunk 1 above)
Comment (mJarsater) [inline components/src/components/dropdown/dropdown.tsx:153]:
```suggestion
```

**Pull request merged**